Remove console logs from Node package

theycallmeswift opened the pull request:
This change fixes #1016 by remove the 4 console log statements from the node package. The only material change here is that workspaces now raise an error (instead of silently failing) if executed in the browser environment, which seems like a saner approach anyway.

theycallmeswift committed:
Remove console logs from Node package

Fixes #1016

tomlinton commented:
There is some additional info on this in https://github.com/project-serum/anchor/issues/532. I think it makes sense to gate it with a flag rather than remove it completely, e.g.

```
        const quiet = features.isSet("quiet-console-errors");
        if (!quiet) console.log("Translating error", err);
```

It's useful to have these by default, and leave it up to the user to turn them off if they want a clean output.

theycallmeswift commented:
@tomlinton I'm happy to make that change, but I would advocate for the opposite default (flag to turn on verbose logging, vs flag to turn off logs) because the developer experience will be better. 

Errors will naturally bubble up to the developer when they are raised. Since we're both throwing the error and logging in all of these cases, we're just ending up with duplicate logs. While an experienced Anchor user may be comfortable with this nuance, it would be confusing for a new user (I can say this from first-hand experience).

Are you ok with my proposal? If so, I'll make the requested changes.

tomlinton commented:
I'm not a maintainer or anything you don't have to defer to me, but I agree that off by default sounds like a good approach. 🙂

theycallmeswift committed:
Enable debugging logs via feature flag

```
const anchor = require("@project-serum/anchor");
anchor.utils.features.set('debug-logs');
```

theycallmeswift commented:
I've updated to enable debug logging by setting the appropriate feature flag.

```
const anchor = require("@project-serum/anchor");
anchor.utils.features.set('debug-logs');
```

Whenever this gets merged, it should close both #532 and #1016.
